tests/integration/test_charm.py

```diff
@@ -321,12 +321,40 @@ def unit_uri(ip_address: str) -> str:
     return "mongodb://{}:{}/".format(ip_address, PORT)
 
 
+def fetch_primary(replica_set_hosts: List[str]) -> str:
+    """Returns IP address of current replica set primary."""
+    # connect to MongoDB client
+    if len(replica_set_hosts) == 1:
+        replica_set_uri = replica_set_hosts[0] + ":" + str(PORT)
+        client = MongoClient(replica_set_uri, replicaset="rs0")
```

Comment:
Could this logic be moved into `generate_replica_set_uri`?

Reply:
I don't see why not 😉